PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = ["project.project"]
-    _inherit = ['project.project.stage']
-    _inherit = ["project.task"]
-    _inherit = ["project.task.type"]

Total Changes: 8

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/project_sms/models/project_project.py
+++ b/addons/project_sms/models/project_project.py
@@ -5,7 +5,7 @@ from odoo import api, models
 
 
 class ProjectProject(models.Model):
-    _inherit = ["project.project"]
+    _inherit = "project.project"
 
     def _send_sms(self):
         for project in self:

--- a/addons/project_sms/models/project_stage.py
+++ b/addons/project_sms/models/project_stage.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class ProjectProjectStage(models.Model):
-    _inherit = ['project.project.stage']
+    _inherit = 'project.project.stage'
 
     sms_template_id = fields.Many2one('sms.template', string="SMS Template",
         domain=[('model', '=', 'project.project')],

--- a/addons/project_sms/models/project_task.py
+++ b/addons/project_sms/models/project_task.py
@@ -5,7 +5,7 @@ from odoo import api, models
 
 
 class ProjectTask(models.Model):
-    _inherit = ["project.task"]
+    _inherit = "project.task"
 
     def _send_sms(self):
         for task in self:

--- a/addons/project_sms/models/project_task_type.py
+++ b/addons/project_sms/models/project_task_type.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class ProjectTaskType(models.Model):
-    _inherit = ["project.task.type"]
+    _inherit = "project.task.type"
 
     sms_template_id = fields.Many2one('sms.template', string="SMS Template",
         domain=[('model', '=', 'project.task')],
